src/deluge/model/song/song.cpp

```diff
@@ -2940,7 +2942,9 @@ void Song::turnSoloingIntoJustPlaying(bool getRidOfArmingToo) {
 		clip->soloingInSessionMode = false;
 
 		if (getRidOfArmingToo) {
-			clip->armState = ArmState::OFF;
+			if (clip->launchStyle != LaunchStyle::FILL && clip->launchStyle != LaunchStyle::ONCE) {
```

Comment:
Same here

Reply:
Same as before, this avoids those two type of clips to be unarmed. If not, the Fill and Once clips will stay playing forever. This make sure that those two launchmode clips will get stopped after one play, and not other launchmodes that we might add in the future